polars/polars-arrow/src/kernels/rolling/no_nulls/mod.rs

```diff
@@ -112,20 +112,15 @@ fn compute_var_weights<T>(vals: &[T], weights: &[T]) -> T
 where
     T: Float + std::ops::AddAssign,
 {
-    let weighted_iter = vals.iter().zip(weights).map(|(x, y)| *x * *y);
-
-    let mut sum = T::zero();
-    let mut sum_of_squares = T::zero();
-
-    for val in weighted_iter {
-        sum += val;
-        sum_of_squares += val * val;
-    }
-    let count = NumCast::from(vals.len()).unwrap();
+    // Assumes the weights have already been standardized to 1
```

Comment:
Can you add a `debug_assert` to assert that assumption?

Reply:
I didn't do that because it's a private function only ever used here and it gets called for every window. So summing and checking the weights each time feels like a lot of wasted work.

Reply:
It is, that's why it is a `debug_assert`. It will only run on debug builds, e.g. when we develop and run tests.